euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/Join.java

```diff
@@ -129,7 +129,10 @@
     private final UnaryFunction<LEFT, KEY> leftKeyExtractor;
     private final UnaryFunction<RIGHT, KEY> rightKeyExtractor;
     private final BinaryFunctor<LEFT, RIGHT, OUT> joinFunc;
-    private boolean outer;
+
+    private boolean outer = false;
+    private boolean leftCached = false;
+    private boolean rightCached = false;
```

Comment:
I would drop this, because it just brings unnecessary implementation complexity and copy&pasting.